### PR TITLE
New era-channel conditioned GNN; dump GNN arrays by default; updates to samples

### PR DESCRIPTION
**Pull request by mondalspandan**

### CODE REVIEW

**Review by andreypz — COMMENTED** [review]
To be consistent with MET to PuppiMET switch, I think we need to modify it also in selectors:
https://github.com/cms-rwth/VHccPoCo/blob/main/CommonSelectors.py


In the future we should make it configurable somehow. When running over Run2 (which we may want to do later) we need to use regular MET

**Comment by mondalspandan** [comment]
That's a good point. I have updated the common selectors. But in the future we'll need a switch and a redefinition like "events.myMET = events.PuppiMET or events.MET"

File: README.md [hunk 1]
```diff
@@ -19,9 +19,9 @@
 	conda install conda-forge::xrootd
 	conda install conda-forge::lightgbm
 	conda install conda-forge::tensorflow
-    conda install pytorch::pytorch
-    conda install conda-forge:alive-progress
-    conda install conda-forge:optuna
+  conda install pytorch::pytorch        # This is not tested to work. If it fails, use `pip install torch`
```
Comment (andreypz) [inline README.md]: I tested this and it worked for me

File: cfg_VHcc_WLNu.py [hunk 2]
```diff
@@ -274,11 +287,11 @@
...
         "top_mass": HistConf( [Axis(field="top_mass", bins=100, start=0, stop=400, label=r"$M_{top}$ [GeV]")] ),
 
-        "BDT": HistConf( [Axis(field="BDT", bins=1000, start=0, stop=1, label="BDT")],
+        "BDT": HistConf( [Axis(field="BDT", bins=10000, start=0, stop=1, label="BDT")],
                          only_categories = ['SR_Wmn_2J_cJ','SR_Wen_2J_cJ']),
-        "DNN": HistConf( [Axis(field="DNN", bins=1000, start=0, stop=1, label="DNN")],
+        "DNN": HistConf( [Axis(field="DNN", bins=10000, start=0, stop=1, label="DNN")],
                          only_categories = ['SR_Wmn_2J_cJ','SR_Wen_2J_cJ']),
-        "GNN": HistConf( [Axis(field="GNN", bins=1000, start=0, stop=1, label="GNN")],
+        "GNN": HistConf( [Axis(field="GNN", bins=10000, start=0, stop=1, label="GNN")],
```
Comment (andreypz) [inline cfg_VHcc_WLNu.py:294]: I can't believe we actually need 10000 bins for optimization... but Ok, we can run with it for now, will review later

File: params/shapemaker_vhcc_run3/Zll.yaml [hunk 3]
```diff
@@ -86,13 +86,13 @@ sample_to_process_map:
   DATA_DoubleMuon: "data_obs"
   ZH_Hto2C_Zto2L: "ZH_hcc"
   ZH_Hto2B_Zto2L: "ZH_hbb"
-  ggZH_Hto2C_Zto2L: "ggZH_hcc"
-  ggZH_Hto2B_Zto2L: "ggZH_hbb"
+  ggZH_Hto2C_Zto2L: "ZH_hcc"
```
Comment (andreypz) [inline params/shapemaker_vhcc_run3/Zll.yaml]: I don't think we want to merge ggZH with ZH.  For two reasons: 1) at the moment merging is not implemented, they will be overwritten. 2) these are two separate processes and will have different systematics from theory. Even in Run2 they were treated separately.

**Review by andreypz — CHANGES_REQUESTED** [review]
Looks good, except for `ggZH` should be separate process for shape maker. in plotting it's OK to merge

**Comment by andreypz** [comment]
Regarding PuppiMet, I created an issue here: https://github.com/PocketCoffea/PocketCoffea/issues/293

File: README.md (hunk 1 above)
Comment (uttiyasarkar) [reply README.md]: The README file looks like it has missing characters

**Review by andreypz — APPROVED** [review]

**Pull request merged**